esphome/components/mitsubishi_uart/muart_button.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "esphome/components/button/button.h"
+#include "mitsubishi_uart.h"
+
+namespace esphome {
+namespace mitsubishi_uart {
+
+class MUARTButton : public button::Button, public Component, public Parented<MitsubishiUART> {
+ public:
+  MUARTButton() = default;
+  using Parented<MitsubishiUART>::Parented;
+
+ protected:
+  virtual void press_action() override;
```

Comment:
This is overriding the method but not defining it, so I think this is a bug (I'm surprised how it works now).

I think you need to either `virtual void press_action() override = 0;`, or... just remove the line altogether? button::Button provides this as a virtual method and you're just inheriting it with no changes here.

Also, FilterResetButton::press_action likely needs an override keyword as well.

Reply:
Another good catch.  Just committed https://github.com/muart-group/esphome/pull/42/commits/129e61b2f9995d48878b2a051bc48f07d427c0aa to #42 to clean that up.

C isn't my first language so I'm not 100% sure, but I think it was still working okay because no one ever called `MUARTButton::press_action()`, only `FilterResetButton::press_action()`.